anomaly-detection/README.MD

```diff
@@ -0,0 +1,211 @@
+
+# Event pattern detection with Apache Flink and Pravega
+
+
+Sample application which simulates network anomaly intrusion and detection using Apache Flink and Apache Pravega.
+
+This application is based on [streaming-state-machine](https://github.com/StephanEwen/flink-demos/tree/master/streaming-state-machine) which is slightly extended to demonstrate Pravega/Flink integration capabilities.
+
+Events in streams (generated by devices and services, such as firewalls login-, and authentication services) are expected to occur in certain patterns. Any deviation from these patterns indicates an anomaly (attempted intrusion) that the streaming system should recognize and that should trigger an alert.
+
+The event patterns are tracked per interacting party (here simplified per source IP address) and are validated by a state machine. The state machine's states define what possible events may occur next, and what new states these events will result in.
+ 
+The final aggregated results are grouped under network id which acts as a network domain abstraction hosting multiple server machines.
+
+The aggregated results are sinked to Elastic Search for visualizing from Kibana user interface.
+
+The following diagram depicts the state machine used in this example.
+
+```
+           +--<a>--> W --<b>--> Y --<e>---+
+           |                    ^         |     +-----<g>---> TERM
+   INITIAL-+                    |         |     |
+           |                    |         +--> (Z)---<f>-+
+           +--<c>--> X --<b>----+         |     ^        |
+                     |                    |     |        |
+                     +--------<d>---------+     +--------+
+```
+
+## Getting Started
+
+
+### Building Pravega & Flink connectors
+
+Follow below steps to build and publish artifacts from source to local Maven repository:
+
+```
+$git clone https://github.com/pravega/pravega.git
+$./gradlew clean build
+$./gradlew install
+
+$git clone https://github.com/pravega/flink-connectors.git
+$./gradlew clean build
+$./gradlew install
+
+```
+Alternatively, follow the instructions from [here](http://pravega.io/docs/getting-started/) to pull from release repository.
+
+### Building and Running Example
+
+- Clone the repository 
+- Build samples `mvn clean build`
+- Create local distribution `mvn clean installDist`
+
+<b>To run the application using script (Embedded Flink App)</b>
+
+Navigate to `$PRAVEGA_SAMPLES_HOME/anomaly-detection/build/install/pravega-flink-anomaly-detection/`
+
+Update `conf/app.json` (see "Application Configuration" section) as appropriate and run below commands to create stream and publish/consume records.
+
+```
+`bin/anomaly-detection --configDir <APP_CONFIG_DIR> --mode <RUN_MODE>`
```

Comment:
Not specifically but initially had plans to support consuming more files from the configuration directory.